vars/cloudformation.groovy

```diff
@@ -75,58 +76,79 @@ def call(body) {
 }
 
 @NonCPS
-def handleActionRequest(cf, config, action){
-  def success = false
-  def result = false // Did something meaningful happen / does it exist?
-
-  switch(action) {
-    case 'exists':
-      if(doesStackExist(cf,config.stackName)) {
-        println "Environment ${config.stackName} already exists"
-        env["${config.stackName}_exists"] = 'true'
-        result = true
-      } else {
-        println "Environment ${config.stackName} does not exist"
-        env["${config.stackName}_exists"] = 'false'
-      }
-      success = true
-      break
-    case 'create':
-      if(!doesStackExist(cf,config.stackName)) {
-        create(cf, config)
-        success = wait(cf, config.stackName, StackStatus.CREATE_COMPLETE)
-        result = true
-      } else {
-        println "Environment ${config.stackName} already Exists"
-        success = true
-      }
-      break
-    case 'delete':
-      delete(cf, config)
-      success = wait(cf, config.stackName, StackStatus.DELETE_COMPLETE)
-      result = true
-      break
-    case 'update':
-      if(update(cf, config)) {
-        success = wait(cf, config.stackName, StackStatus.UPDATE_COMPLETE)
-        result = true
-      } else {
-        success = true
-      }
-      break
-    case 'createOrUpdate':
-      result = handleActionRequest(cf, config, 'create')
-      if (!result) {
-        result = handleActionRequest(cf, config, 'update')
-      }
-      success = true // Non-successes would have already thrown an exception in recursive calls
+def handleActionRequest(cf, config){
+  def assertOrThrow = { booleanAssertion ->
+    if (!booleanAssertion) {
+      printFailedStackEvents(cf, config.stackName, config.region)
+      throw new Exception("Stack ${config.stackName} failed to ${config.action}")
+    }
+  }
+  
+  // Should start an action
```

Comment:
There is a confusing mixture of:
- setting the `env` variables
- printing status
- returning

Depending on action and states. Seems messy and it is hard to figure out if these differences are intentional and/or important.

Reply:
Yes, it kind of started that way though. If we'd want to clean that up, we'd probably have to redo this a bit and then not have it be compatible with the upstream anymore?